src/metabase/pulse.clj

```diff
@@ -80,29 +86,73 @@
     (compare (:row dashcard-1) (:row dashcard-2))
     (compare (:col dashcard-1) (:col dashcard-2))))
 
-(defn- virtual-card-of-type?
-  "Check if dashcard is a virtual with type `ttype`.
+(defn virtual-card-of-type?
+  "Check if dashcard is a virtual with type `ttype`, if `true` returns the dashcard, eles returns `nil`.
```

Comment:
```suggestion
  "Check if dashcard is a virtual with type `ttype`, if `true` returns the dashcard, else returns `nil`.
```

Reply:
Why does this return the dashcard if `true`, and not just `true`? If a function name ends with `?` it should return a boolean by convention